src/Instrumentation/Laravel/src/Watchers/LogWatcher.php

```diff
@@ -37,7 +37,9 @@ public function recordLog(MessageLogged $log): void
         $underlyingLogger = $this->logger->getLogger();
 
         /** @phan-suppress-next-line PhanUndeclaredMethod */
-        if (method_exists($underlyingLogger, 'isHandling') && !$underlyingLogger->isHandling($log->level)) {
+        if (method_exists($underlyingLogger, 'isHandling') && !$underlyingLogger->isHandling(
+                $underlyingLogger->toMonologLevel($log->level))
```

Comment:
this seems like it would now _only_ work with monolog as the underlying logger, but there could be others, right? is `toMonologLevel` a part of any common interface between loggers?

Reply:
@brettmc 
I'm not quite sure how many of the third-party libraries in the industry that comply with `\Psr\Log\LoggerInterface` support the method `method_exists($underlyingLogger, 'isHandling')`.

- Suppose there are many. Then I wonder if adding `method_exists($underlyingLogger, 'toMonologLevel')` is too specific.
- Suppose there are few, and it is more for supporting `Monolog`. Then I think it is acceptable to add `method_exists($underlyingLogger, 'toMonologLevel')`.

Of course, there is another way, which is to add an adapter to solve this problem, but the extension might be a bit cumbersome.

So, I'd like to know your thoughts.

Reply:
Also, there is another method, that is, I define an interface of ShouldHandle and bring it in when doing `new LogWatcher($this->instrumentation)`. It's probably like this:

```php
<?php

declare(strict_types=1);

use Illuminate\Log\Events\MessageLogged;
use Illuminate\Log\LogManager;

interface ShouldHandle
{
    public function shouldHandle(LogManager $logger, MessageLogged $log): bool;
}
```

Reply:
@brettmc 
Another processing method. Because` method_exists($underlyingLogger, 'isHandling')` it is too special. This method is not supported in `\Psr\Log\LoggerInterface`. 

As an official extension package, for this special case, I suggest that users can consider overriding the `recordLog` method of `LogWatcher`.

And for this package, we only need to add a `shoudRecordLog` method.

----

For this, what we may have to do:

- Roll back https://github.com/open-telemetry/opentelemetry-php-contrib/pull/264
- Add a `shoudRecordLog` method 

> Of course, there are some other additions. That is, it can be retrieved more conveniently through some configurations. Currently, our register entry has some limitations. But this may take some time to continuously improve.